### PR TITLE
Cfnlirk3

### PR DESCRIPTION
**Pull request by kamalojasv181**

### CODE REVIEW

**Comment by kamalojasv181** [comment]
https://github.com/SciML/OrdinaryDiffEq.jl/issues/1093

File: src/perform_step/kencarp_kvaerno_perform_step.jl [hunk 1]
```diff
@@ -412,6 +414,213 @@ end
...
+  ################################## Solve Step 3
+
+  if typeof(integrator.f) <: SplitFunction
+    z₃ .= z₂
+    @.. u = tmp + γ*z₂
+    f2(k2, u, p, t + c2*dt); k2 .*= dt
+    integrator.destats.nf2 += 1
+    @.. tmp = uprev + a31*z₁ + a32*z₂ + ea31*k1 + ea32*k2
+  else
+    # Guess is from Hermite derivative on z₁ and z₂
```
Comment (ChrisRackauckas) [inline src/perform_step/kencarp_kvaerno_perform_step.jl:582]: how is this from Hermite?

Comment (kamalojasv181) [reply src/perform_step/kencarp_kvaerno_perform_step.jl]: Oops!. Removed the comment.

**Comment by ChrisRackauckas** [comment]
Does the paper describe the stage predictors? How did you make the choice here?

**Comment by kamalojasv181** [comment]
Nope. It doesn't .  I just thought initializing it to a close value might help it converge it faster

**Comment by ChrisRackauckas** [comment]
Yes, that's known as a minimum correction predictor

**Comment by kamalojasv181** [comment]
So, I just take the previous stage values as predictors. Is there anything better I could have done?

**Comment by ChrisRackauckas** [comment]
It's good enough for now. We should open an issue afterwards to investigate other stage predictors for this. You can do things like derive interpolation estimates for a `c_i` ever decreases, since now you're not extrapolating. Stuff like that can improve the implementation quite a bit.

**Comment by kamalojasv181** [comment]
Can you please give me a relevant pointer? I really need to catch up before opening an issue.

**Comment by kamalojasv181** [comment]
Anything else besides the predictors?

**Comment by ChrisRackauckas** [comment]
If you haven't read https://www.sciencedirect.com/science/article/abs/pii/0168927495001158 it's a good overview to implementations of stiff ODE solvers.

**Comment by kamalojasv181** [comment]
I think I should open up a new issue. Because I have done the same in bunch of other places as well. Alright?

**Comment by ChrisRackauckas** [comment]
Just in the SDIRK methods. Yeah, we should get a list of the ones you did that might need a second look at the stage predictors.

Looks like the tests are failing here though?

**Comment by kamalojasv181** [comment]
Nope. Algconvergence 2 is failing. I havent added any tests that are a part of Algconvergence 2

**Comment by ChrisRackauckas** [comment]
Looks like compile times may have increased when it updated to Julia v1.4. I restarted the test, but ping me tomorrow if it doesn't pass. I think it can just be broken up or some print statements can be added to stop Travis from timing out.

**Comment by ChrisRackauckas** [comment]
Fixed on master. Rebase.

**Pull request merged**